NICE.Identity.Management/ClientApp/src/views/User/User.tsx

```diff
@@ -1,20 +1,101 @@
-import React from "react";
+import React, { Component } from "react";
+
 import { RouteComponentProps, Link } from "react-router-dom";
+import { Breadcrumbs, Breadcrumb } from "@nice-digital/nds-breadcrumbs";
+import { Panel } from "@nice-digital/nds-panel";
+import { Tag } from "@nice-digital/nds-tag";
+
+import { UserType } from "../../models/types";
+import { Endpoints } from "../../data/endpoints";
+import { UnlockUser } from "../../components/UnlockUser/UnlockUser";
 
 type TParams = { id: string };
 
 type UserProps = {} & RouteComponentProps<TParams>;
 
-export const User = (props: UserProps) => {
-	const { match } = props;
-	const { params } = match;
-
-	return (
-		<div className="grid">
-			<div data-g="12">
-				<p>User: {params.id}</p>
-				<Link to={`/users/${params.id}/roles`}>Roles</Link>
-			</div>
-		</div>
-	);
+type UserState = {
+	data: UserType;
+	error: string;
+	isLoading: boolean;
 };
+
+export class User extends Component<UserProps, UserState> {
+	constructor(props: UserProps) {
+		super(props);
+
+		this.state = {
+			data: {} as UserType,
+			error: "",
+			isLoading: true,
+		};
+	}
+
+	handleError = (error: Error) => {
+		this.setState({ error: error.message });
+	};
+
+	updateData = (updatedData: UserType) => {
+		this.setState({ data: updatedData });
+	};
+
+	fetchData = async (url: string) => {
+		try {
+			const response = await fetch(url);
+			const data = await response.json();
+			this.setState({ data, isLoading: false });
+		} catch (error) {
+			this.setState({ error });
```

Comment:
This seems to set the whole error object, but the state is a string? I think an optional error object for the state makes more sense than a string. An error object has everything that's useful that we can use for showing a developer in dev mode for example.

ALSO, shouldn't this set `isLoading: false`?